R/estimate_KM.R

```diff
@@ -65,27 +69,27 @@
 #'  )
 #'
 #' visR::estimate_KM(data = veteran_adam, strata = "trt")
+NULL
 
-estimate_KM <- function(
+#' @rdname estimate_KM
+#' @export
+estimate_KM.data.frame <- function(
     data = NULL
     ,strata = NULL
     ,CNSR = "CNSR"
     ,AVAL = "AVAL"
     ,...
 ){
-
-  # Capture input to validate user input for data argument ---------------------
-
+  # if user passed a formula, dispatch via estimate_KM.formula -----------------
   dots <- rlang::dots_list(...)
+  if ("formula" %in% names(dots)) {
+    formula <- dots[["formula"]]
+    dots[["formula"]] <- NULL
+    message("To use the `estimate_KM(formula=)` argument, place it in the first position.")
+    return(rlang::inject(estimate_KM.formula(formula = formula, data = data, !!!dots)))
+  }
 
   # Validate argument inputs ---------------------------------------------------
-
```

Comment:
Why is this removed? Is it automatically handled by the method? If yes the testing needs to be updated.

Reply:
Yes, auto-handled by the S3 method.